aiida_pseudo/groups/family/pseudodojo.py

```diff
@@ -0,0 +1,227 @@
+# -*- coding: utf-8 -*-
+"""Subclass of `PseudoPotentialFamily` designed to represent an PseudoDojo configuration."""
+from collections import namedtuple
+from typing import Sequence
+
+from .pseudo import PseudoPotentialFamily
+from .psml import PsmlFamily
+from .psp8 import Psp8Family
+from .upf import UpfFamily
+
+__all__ = (
+    'PseudoDojoConfiguration', 'PseudoDojoFamily', 'PseudoDojoPsp8Family', 'PseudoDojoUpfFamily', 'PseudoDojoPsmlFamily'
+)
+
+PseudoDojoConfiguration = namedtuple(
+    'PseudoDojoConfiguration', ['version', 'functional', 'rel', 'protocol', 'hint', 'format']
+)
+
+
+class PseudoDojoFamily(PseudoPotentialFamily):
+    """Subclass of `PseudoPotentialFamily` designed to represent an PseudoDojo configuration.
+
+    The `PseudoDojoFamily` is essentially a `PseudoPotentialFamily` with some additional constraints.
+    It can only be used to contain the pseudo potentials and corresponding metadata of an official
+    PseudoDojo configuration.
+    """
+
+    label_template = 'PseudoDojo/{version}/{functional}/{rel}/{protocol}/{hint}/{format}'
+
+    @classmethod
+    def get_valid_labels(cls) -> Sequence[str]:
+        """Return the tuple of labels of all valid PseudoDojo configurations."""
+        return tuple(cls.format_configuration_label(configuration) for configuration in cls.valid_configurations)
+
+    @classmethod
+    def format_configuration_label(cls, configuration: PseudoDojoConfiguration) -> str:
+        """Format a label for an `PseudoDojoFamily` with the required syntax.
+
+        :param configuration: the PseudoDojo configuration
+        :return: label
+        """
+        return cls.label_template.format(
+            version=configuration.version,
+            functional=configuration.functional,
+            rel=configuration.rel,
+            protocol=configuration.protocol,
+            hint=configuration.hint,
+            format=configuration.format
+        )
+
+    def __init__(self, label=None, **kwargs):
+        """Construct a new instance, validating that the label matches the required format."""
+        if label not in self.get_valid_labels():
+            raise ValueError(f'the label `{label}` is not a valid PseudoDojo configuration label.')
+
+        super().__init__(label=label, **kwargs)
+
+
+class PseudoDojoPsp8Family(PseudoDojoFamily, Psp8Family):
+    """Subclass of `PseudoDojoFamily` and `Psp8Family` designed to represent a psp8 PseudoDojo configuration.
+
+    The `PseudoDojoPsp8Family` is essentially a `PseudoDojoFamily` with some additional constraints.
+    It can only be used to contain the pseudo potentials and corresponding metadata of an official psp8
+    PseudoDojo configuration.
+    """
+
+    default_configuration = PseudoDojoConfiguration('04', 'pbe', 'sr', 'standard', 'normal', 'psp8')
+    valid_configurations = (
+        # nc-sr-03_pbe
+        PseudoDojoConfiguration('03', 'pbe', 'sr', 'standard', 'none', 'psp8'),
```

Comment:
Should `'none'` not just be `None`? Or does the string `'none'` have a special meaning for Pseudo Dojo?

Reply:
`None` is fine, I wasn't sure if I should keep all entries in the named tuple as strings.